### PR TITLE
Additional errors and warnings

### PR DESCRIPTION
**Pull request by GerjonM**
Added the error and warninghook, which has been implemented in FieldAccessors already, to the other accessors - SubForm, RepeatingForm, RepeatingIndexedForm, and the Form itself.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 0.20.1
+
+-   Added a warning and error hook to the `Subform`, `RepearingForm`,
```
Comment (faassen) [inline CHANGES.md]: RepearingForm

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+    and `Form` accessors. These hooks are accesible the same way as
```
Comment (faassen) [inline CHANGES.md]: Accessible

File: README.md [hunk 3]
```diff
@@ -609,6 +609,10 @@ contains both an internally generated error message and one that is generated
 via `getError`, the internally generated message trumps the one returned by the
 `getError` hook.
 
+Other accessors in mstform - `SubForm`, `RepeatingForm` and `Form` - also contain
+this error hook, allowing you to set errors on the complete form - or any accessor
```
Comment (faassen) [inline README.md]: They don't contain the error hook. They use the error hook.

File: src/sub-form-accessor.ts [hunk 4]
```diff
@@ -25,6 +25,10 @@ export class SubFormAccessor<
     // no op
   }
 
+  async validate(): Promise<boolean> {
+    return this.errorValue === undefined;
```
Comment (faassen) [inline src/sub-form-accessor.ts]: This doesn't look right: a sub form is also invalid if any of its fields or other sub-accessors is invalid. Why did you need to add this?

File: test/form.test.ts [hunk 5]
```diff
@@ -730,52 +730,6 @@ test("setErrors", async () => {
   expect(field.error).toEqual("WRONG");
 });
 
-test("setErrors repeating", async () => {
```
Comment (faassen) [inline test/form.test.ts]: This test shouldn't be removed. It tests that the bar field in a repeating form can have an error set, which should still be possible.

File: test/form.test.ts [hunk 6]
```diff
@@ -730,52 +730,6 @@ test("setErrors", async () => {
...
-  const o = M.create({ foo: [{ bar: "FOO" }] });
-
-  const state = form.state(o);
-  state.setErrors({ foo: [{ bar: "WRONG" }] });
-
-  const field = state.repeatingForm("foo").accessors[0].field("bar");
-  expect(field.error).toEqual("WRONG");
-});
-
-test("setErrors directly on repeating", async () => {
```
Comment (faassen) [inline test/form.test.ts]: This test is okay to remove if we drop support for this.

File: CHANGES.md (hunk 1 above)
Comment (faassen) [reply CHANGES.md]: You made more changes than this. You also made it impossible to set errors directly on repeating forms and such apart from the hook. Since that's a major change in behavior you should at least mention it.

File: src/sub-form-accessor.ts (hunk 4 above)
Comment (GerjonM) [reply src/sub-form-accessor.ts]: I added this to determine the validity of the subForm. Without it, the validate function form state.ts would not kick this off.

You do have a point that the subform is also invalid if some of its sub-accessors are. I will ad them to this function

File: test/form.test.ts (hunk 5 above)
Comment (GerjonM) [reply test/form.test.ts]: I will put it back

File: CHANGES.md [hunk 7]
```diff
@@ -1,3 +1,12 @@
+# 0.20.1
```
Comment (faassen) [inline CHANGES.md]: I think this should be release 0.21 as we break backwards compatibility. That's still not proper semantic versioning but at least it's clearer that we break things than a .1 release, which is really reserved for bugfixes only.

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "mstform",
-    "version": "0.19.0",
+    "version": "0.20.0",
```
Comment (faassen) [inline package.json]: Should be 0.21
